CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2440,6 +2440,8 @@ protected function completeTransaction($result, $contributionID) {
             'is_transactional' => FALSE,
             'fee_amount' => CRM_Utils_Array::value('fee_amount', $result),
             'receive_date' => CRM_Utils_Array::value('receive_date', $result),
+            'card_type_id' => CRM_Core_BAO_FinancialTrxn::formatCreditCardDetails(CRM_Utils_Array::value('credit_card_type', $result)),
```

Comment:
we shouldn't need this if we call beginPostProcess (or similar as I think this form doesn't share that )